src/HID-APIs/DefaultKeyboardAPI.hpp

```diff
@@ -42,26 +42,53 @@ size_t DefaultKeyboardAPI::set(KeyboardKeycode k, bool s)
 	}
 	// Its a normal key
 	else{
-		// Add k to the key report only if it's not already present
-		// and if there is an empty slot. Remove the first available key.
-		for (uint8_t i = 0; i < sizeof(_keyReport.keycodes); i++)
-		{
-			auto key = _keyReport.keycodes[i];
-			
-			// Is key already in the list or did we found an empty slot?
-			if (s && (key == uint8_t(k) || key == KEY_RESERVED)) {
-				_keyReport.keycodes[i] = k;
-				return 1;
+		// get size of keycodes during compile time
+		const uint8_t keycodesSize = sizeof(_keyReport.keycodes);
```

Comment:
Yeah, I just followed the original code style because I did not want to make disruptive changes. I do recognise that changing the data type of the subject in question would mess it up. Like you say, HID spec scancodes won't be changing sizes any time soon so this should not matter but it's a good habit to always do it regardless anyway especially considering that this is calculated in compile time anyway so its not like checking `sizeof(array)/sizeof(array[0])` is slower than the other way.